Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -87,11 +87,39 @@ std::vector<std::string> splitSmartsIntoComponents(
   }
   return res;
 }
+
+ROMol* constructMolFromString(const std::string &txt,
+    std::map<std::string,std::string> *replacements, bool useSmiles, bool sanitize) {
+  ROMol *mol;
+  if(!useSmiles){
+    mol=SmartsToMol(txt,0,false,replacements);
+    if(sanitize){
+      try{
+        mol->updatePropertyCache();
+        RingInfo *ringInfo=mol->getRingInfo();
+        if(!ringInfo->isInitialized()){
+          ringInfo->initialize();
+        }
+      }
+      catch (const MolSanitizeException &){
+        std::cerr << "Molecule could not be sanitized" << std::endl;
+      }
+    }
+  }
+  else{
+    mol=SmilesToMol(txt,0,sanitize,replacements);
+  }
+  return mol;
+}
+
+
+
+
 }  // end of namespace DaylightParserUtils
 
 ChemicalReaction *RxnSmartsToChemicalReaction(
     const std::string &text, std::map<std::string, std::string> *replacements,
-    bool useSmiles) {
+    bool useSmiles, bool sanitize) {
```

Comment:
Why is sanitize an option?  We really have no SanitizeRXN(...) function (actually I am submitting one for the next release) so why would we not always do this?

You are also going to run into issues with the difference between sanitizing "smiles" and sanitizing smarts, this is going to fail for many, many smarts patterns.

What I had fallen back to was

```
def SanitizeReactant(mol, mergeUnmappedOnly=True):
    # fixup aromaticity
    Chem.SanitizeMol(mol, Chem.SanitizeFlags.SANITIZE_SETAROMATICITY)
    # merge any unmapped hydrogens
    template = rdmolops.MergeQueryHs(mol, mergeUnmappedOnly=mergeUnmappedOnly)
    return template
```